app/models/image_import_job/dispatcher.rb

```diff
@@ -0,0 +1,40 @@
+class ImageImportJob
+  class Dispatcher < Job::Dispatcher
+    def dispatch
+      pending, = Benchmark.realtime_block(:pending_import_jobs) { pending_import_jobs }
+      running, = Benchmark.realtime_block(:running_import_jobs) { running_import_jobs }
+      busy_src_ids = running.flat_map { |_, dst| dst.map { |job| job.options[:src_provider_id] } }.uniq
+
+      pending.each do |dst_id, pending_jobs|
+        break if running[dst_id].present? || busy_src_ids.include?(dst_id)
+
+        pending_job = pending_jobs.detect { |x| busy_src_ids.exclude?(x.options[:src_provider_id]) }
+
+        unless pending_job.nil?
+          busy_src_ids << pending_job.options[:src_provider_id]
+          do_dispatch(pending_job)
+        end
```

Comment:
Minor but typically prefer `next` rather than having the whole rest of the block in a conditional like:
```suggestion
        pending_job = pending_jobs.detect { |x| busy_src_ids.exclude?(x.options[:src_provider_id]) }
        next if pending_job.nil?

        busy_src_ids << pending_job.options[:src_provider_id]
        do_dispatch(pending_job)
```